projects/packages/publicize/src/class-publicize-utils.php

```diff
@@ -120,4 +120,36 @@ public static function assert_is_wpcom( $method ) {
 	public static function should_use_jetpack_module_endpoint() {
 		return class_exists( 'Jetpack' ) && defined( 'JETPACK__VERSION' ) && ( version_compare( (string) JETPACK__VERSION, '14.3', '>=' ) );
 	}
+
+	/**
+	 * Log a warning that a deprecated endpoint was called.
+	 *
+	 * @param string $function_name        The function name.
+	 * @param string $version              The version in which the endpoint was deprecated.
+	 * @param string $deprecated_endpoint  The deprecated endpoint.
+	 * @param string $alternative_endpoint The alternative endpoint.
+	 */
+	public static function endpoint_deprecated_warning( $function_name, $version, $deprecated_endpoint, $alternative_endpoint = '' ) {
+
+		$messages = array(
+			sprintf(
+				/* translators: %s: REST API endpoint. */
+				esc_html__( '%1$s endpoint has been deprecated.', 'jetpack-publicize-pkg' ),
+				'"' . $deprecated_endpoint . '"'
+			),
+		);
+
+		if ( ! empty( $alternative_endpoint ) ) {
+			$messages[] = sprintf(
+				/* translators: %s: alternative endpoint. */
+				esc_html__( 'Please use %s endpoint instead.', 'jetpack-publicize-pkg' ),
+				'"' . $alternative_endpoint . '"'
+			);
+		}
```

Comment:
What about having two fuller statements here?

'"%1$s" endpoint has been deprecated.'

'"%1$s" endpoint has been deprecated, in favor of "%2$s"'

The translations can deal with it as a whole, rather than concatenating the strings, although I'm not sure about the upgrade message. We could add that at the start. "Please ugrade your plugins - "%1$s" endpoint has been..."

Reply:
It's possible here that an endpoint doesn't have an alternative, so, that `in favor of` may not make sense.